lighthouse-extension/app/src/lighthouse-ext-background.js

```diff
@@ -70,117 +47,101 @@ function updateBadgeUI(optUrl) {
 }
 
 /**
- * @param {!Object} options Lighthouse options.
- * @param {!Array<string>} categoryIDs Name values of categories to include.
- * @return {!Promise}
+ * @param {{flags: LH.Flags}} options Lighthouse options.
+ * @param {Array<string>} categoryIDs Name values of categories to include.
+ * @return {Promise<LH.RunnerResult|void>}
  */
-window.runLighthouseInExtension = function(options, categoryIDs) {
+async function runLighthouseInExtension(options, categoryIDs) {
   // Default to 'info' logging level.
   log.setLevel('info');
   const connection = new ExtensionProtocol();
   options.flags = Object.assign({}, options.flags, {output: 'html'});
 
-  // return enableOtherChromeExtensions(false)
-  // .then(_ => connection.getCurrentTabURL())
-  return connection.getCurrentTabURL()
-    .then(url => window.runLighthouseForConnection(connection, url, options,
-      categoryIDs, updateBadgeUI))
-    .then(runnerResult => {
-      // return enableOtherChromeExtensions(true).then(_ => {
-      const blobURL = window.createReportPageAsBlob(runnerResult, 'extension');
-      return new Promise(resolve => chrome.windows.create({url: blobURL}, resolve));
-      // });
-    }).catch(err => {
-      // return enableOtherChromeExtensions(true).then(_ => {
-      throw err;
-      // });
-    });
-};
+  const url = await connection.getCurrentTabURL();
+  const runnerResult = await background.runLighthouseForConnection(connection, url, options,
+    categoryIDs, updateBadgeUI);
+  if (runnerResult) {
+    // For now, should always be a runnerResult as the extension can't do `gatherMode`
```

Comment:
should we throw if no runner result? seems like a fatal condition

Reply:
SG